client-side/C#/StreetSmartArcGISPro/StreetSmartArcGISPro/Overlays/Measurement/MeasurementList.cs

```diff
@@ -186,15 +188,76 @@ public async Task CreateMeasurement(ArcGISGeometryType geometryType)
             measurement2.IsDisposed = false;
             FromMap = true;
 
-            //IMeasurementOptions options = MeasurementOptionsFactory.Create(measurementGeometryType);
-            //creates the save measurement button
             IMeasurementOptions options = MeasurementOptionsFactory.Create(measurementGeometryType, MeasureMethods.DepthMap, true);
-            await Api.StartMeasurementMode(panoramaViewer, options);
+            try
+            {
+              string epsgCode = CoordSystemUtils.CheckCycloramaSpatialReferenceMapView(MapView.Active);
+              string srsUnit = GetSrsUnit(epsgCode);
+
+              bool isMeasurementNotAllowed = (srsUnit.Equals("degree", StringComparison.OrdinalIgnoreCase) || srsUnit.Equals("degrees", StringComparison.OrdinalIgnoreCase) || srsUnit.Equals("Unknown unit", StringComparison.OrdinalIgnoreCase) || srsUnit.Contains("Error", StringComparison.OrdinalIgnoreCase));
+              if (!isMeasurementNotAllowed)
+              {
+                await Api.StartMeasurementMode(panoramaViewer, options);
+              }
+            }
+            catch (Exception ex)
+            {
+              EventLog.Write(EventLog.EventType.Error, $"Street Smart: (MeasurementList.cs) (CreateMeasurement) {ex.Message}");
+            }
           }
         }
       }
     }
 
+    public static string GetSrsUnit(string srsString)
```

Comment:
Please, make this method returning an enum, not string. Double check if these units already exist in source code.